app/uk/gov/hmrc/thirdpartyapplication/models/db/ApplicationData.scala

```diff
@@ -41,7 +41,8 @@ case class ApplicationData(id: UUID,
                            rateLimitTier: Option[RateLimitTier] = Some(BRONZE),
                            environment: String = Environment.PRODUCTION.toString,
                            checkInformation: Option[CheckInformation] = None,
-                           blocked: Boolean = false) {
+                           blocked: Boolean = false,
+                           cidrBlocks: Set[String] = Set.empty) {
```

Comment:
Does this need to be `Optional[Set[String]]`? Think there are two cases where there are no CIDR blocks defined - where the API requires it but it hasn't been entered (`Some(Set.empty))`), or where the API does not require them (`None`)

Reply:
Why would we need that distinction?

Reply:
Does an empty Set mean that a request should be allowed, or not?

Reply:
Yeah, it should be allowed. But maybe you shouldn't be able to subscribe to an API that requires IP whitelisting if you don't have any IPs configured

Reply:
There are two distinct scenarios - if an API requires whitelisted IPs, and you haven't provided any, that should be a deny. If the API does not require whitelisting, then it should be an allow. So an empty set potentially has two different meanings...

Reply:
If an API suddenly starts requiring whitelisting, would you have to go through all apps subscribed to it and change all empty sets to none values or something? I'm not sure how that would work.

If someone (the gateway for instance) needs to know if the API requires whitelisting maybe they should ask API definition...

Reply:
Yeah, maybe it doesn't matter too much. Would need to check whether the API required it or not first anyway. And if it doesn't, you'd never need to look at the Set.